auth/template.yaml

```diff
@@ -132,6 +172,82 @@ Resources:
       ResourceArn: !GetAtt CognitoUserPool.Arn
       WebACLArn: !GetAtt WebApplicationFirewall.Arn
 
+  WAFLogGroup:
+    Type: AWS::Logs::LogGroup
+    Properties:
+      DataProtectionPolicy:
+        Name: CloudWatchLogs-PersonalInformation-Protection
+        Description: Protect basic types of sensitive data
+        Version: "2021-06-01"
+        Statement:
+          - Sid: audit-policy
+            DataIdentifier:
+              - arn:aws:dataprotection::aws:data-identifier/EmailAddress
+            Operation:
+              Audit:
+                FindingsDestination: {}
+          - Sid: redact-policy
+            DataIdentifier:
+              - arn:aws:dataprotection::aws:data-identifier/EmailAddress
+            Operation:
+              Deidentify:
+                MaskConfig: {}
+      LogGroupClass: STANDARD
+      LogGroupName: !Sub "aws-waf-logs-${AWS::StackName}"
+      KmsKeyId: !GetAtt WAFLogGroupKMSKey.Arn
+      RetentionInDays: 30
+      Tags:
+        - Key: Product
+          Value: GOV.UK
+        - Key: Environment
+          Value: !Ref Environment
+        - Key: System
+          Value: Authentication
+
+  WAFLoggingConfiguration:
+    Type: AWS::WAFv2::LoggingConfiguration
+    Properties:
+      LogDestinationConfigs:
+        - !GetAtt WAFLogGroup.Arn
+      ResourceArn: !GetAtt WebApplicationFirewall.Arn
+
+  WAFLogGroupKMSKey:
```

Comment:
For my understanding, is this required to be able to create logs for WAF?

Reply:
@pazsaragi which part? The KMS?

Reply:
Oh, I think I know what you mean - yeah, you have to explicitly create a log group, it used to be that WAF only wrote to firehose, it's since become more flexible I believe based on the reading I did while creating this resource.